flink-ml-parent/flink-ml-api/src/main/java/org/apache/flink/ml/api/misc/param/Params.java

```diff
@@ -44,17 +86,39 @@
 	 * @param <V>  the type of the specific parameter
 	 * @return the value of the specific parameter, or default value defined in the {@code info} if
 	 * this Params doesn't contain the parameter
-	 * @throws RuntimeException if the Params doesn't contains the specific parameter, while the
-	 *                          param is not optional but has no default value in the {@code info}
+	 * @throws IllegalArgumentException if the Params doesn't contains the specific parameter, while the
+	 *                          param is not optional but has no default value in the {@code info} or
+	 *                          if the Params contains the specific parameter and alias, but has more
+	 *                          than one value or
+	 *                          if the Params doesn't contains the specific parameter, while the ParamInfo
+	 *                          is optional but has no default value
 	 */
-	@SuppressWarnings("unchecked")
 	public <V> V get(ParamInfo<V> info) {
-		V value = (V) paramMap.getOrDefault(info.getName(), info.getDefaultValue());
-		if (value == null && !info.isOptional() && !info.hasDefaultValue()) {
-			throw new RuntimeException(info.getName() +
-				" not exist which is not optional and don't have a default value");
+		String value = null;
+		String usedParamName = null;
+		for (String nameOrAlias : getParamNameAndAlias(info)) {
+			if (params.containsKey(nameOrAlias)) {
+				if (usedParamName != null) {
+					throw new IllegalArgumentException(String.format("Duplicate parameters of %s and %s",
```

Comment:
`usedParamName` is the first found in the list returned by `getParamNameAndAlias(info)`, so when the map contain the param name or the alias more than one, it will throw exception.

Reply:
oh, thanks for the explanation.

Reply:
Could you please add test case for check it?

Reply:
added to `o.a.f.ml.api.misc.ParamsTest$testGetAliasParam`

Reply:
Thanks